Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -84,65 +84,118 @@ void Digitizer::setSimulationParameters()
-    mSignalsMapCollection[det].clear();
-    mDigitsCollection[det].clear();
+  int count = 0; //  counter of the digitized signals from the deque
+  SignalContainer fullSignal;
+  for (int i = 0; i < mPileupSignals.size(); ++i) {
+    const auto& collection = mPileupSignals[i];
+    for (int det = 0; det < kNdet; ++det) {
+      const auto& signalMap = collection[det]; //--> a map with active pads only for this chamber
+      for (const auto& signal : signalMap) {   // loop over active pads only, if there is any
+        const int& key = signal.first;
+        const SignalArray& signalArray = signal.second;
+        // check if the signal is from a previous event
+        if (signalArray.firstTBtime < mCurrentTriggerTime) {
+          if ((mCurrentTriggerTime - signalArray.firstTBtime) < (READOUT_TIME + BUSY_TIME)) {
+            continue; // ignore the signal if it  is too old.
+          }
+          // add only what's leftover from this signal
+          int idx = (int)((mCurrentTriggerTime - signalArray.firstTBtime) * 0.01); // number of bins to add, from the tail
+          auto it0 = signalArray.signals.begin() + idx;
+          auto it1 = fullSignal[key].signals.begin();
+          while (it0 < signalArray.signals.end()) {
+            *it1 += *it0;
+            it0++;
+            it1++;
+          }
+        } else {
+          // the signal is from a triggered event
+          int idx = (int)((signalArray.firstTBtime - mCurrentTriggerTime) * 0.01); // number of bins to add, on the tail of the final signal
+          auto it0 = signalArray.signals.begin();
+          auto it1 = fullSignal[key].signals.begin() + idx;
+          while (it1 < fullSignal[key].signals.end()) {
+            *it1 += *it0;
+            it0++;
+            it1++;
+          }
+          if (it0 < signalArray.signals.end()) {
+            count++; // add one more element to keep from the tail of the deque
+          }
+        }
+
+        // do we need to set this for further processing?, it is ok for setting up the full-signal map
+        fullSignal[key].firstTBtime = mCurrentTriggerTime;
+
+        // keep the labels
+        for (const auto& label : signalArray.labels) {
+          (fullSignal[key].labels).push_back(label); // maybe check if the label already exists? is that even possible?
+        }
+      } // loop over active pads in detector
+    }   // loop over detectors
+  }     // loop over pileup container
+
+  // once the full signal for this chamber is built, one can make the adcs
+  if (fullSignal.size() > 0) {
+    // convert signals to ADCs and fill the digit container
+    bool status = convertSignalsToADC(0, fullSignal, digits);
+    if (!status) {
+      LOG(WARN) << "TRD conversion of signals to digits failed";
+    }
+    // get the labels attached to the signals and put it in the label container
+    for (const auto& iter : fullSignal) {
+      labels.addElements(labels.getIndexedSize(), iter.second.labels);
+    }
+  }
```

Comment:
@sawenzel, calling for help here:

If a signal forms after adding all the signals listed for pileup, a digit is produced. Note that `fullSignal` is a container of signals. In a second step, all the MC labels attached to the signal container are dumped to the label container that is sent out to the DPL. At this point, I am not sure if I am using the MCTruthContainer correctly. I am not getting the pileup and embedding analysis passing, but the ADC signal the truly added.

Reply:
Since the test is passing, I guess the implementation is correct.